### PR TITLE
[promtail] Fix scrape configs

### PR DESCRIPTION
**Pull request by unguiculus**
The default config was missing a few drop actions. This had caused logs
to be added twice, once the correct way and once with an incorrect job
label.

In order to help debug the config, the flag
`config.snippets.addScrapeJobLabel` is introduced. If enabled, an
additional label `scrape_job` is added which identifies the scrape job
that picked up the log message.

### CODE REVIEW

File: charts/promtail/values.yaml [hunk 1]
```diff
@@ -278,10 +283,19 @@ config:
             source_labels:
               - __meta_kubernetes_pod_label_app_kubernetes_io_name
             target_label: app
+          - action: drop
+            regex: ''
+            source_labels:
+              - app
```
Comment (torstenwalter) [inline charts/promtail/values.yaml:289]: are these changes intended? I don't see how this relates to the `addScrapeJobLabel` option.

Comment (unguiculus) [reply charts/promtail/values.yaml:289]: Yes, that's what this PR is mostly about. We need to drop out here when there is no app label in order to avoid duplicates with an incomplete job label. These pods will be covered by one of the other scrape jobs.

That's not related to `addScrapeJobLabel`.

**Review by torstenwalter — APPROVED** [review]
